### ISSUE TITLE
[3.x]: Error in "globals" cp are in sites with a dash "-" in them

### ISSUE THREAD

**Issue by eduardoalonsoalbella** [craftcms/cms#12529]
### Description
 When accesing "globals" with a site with a dash, an error is thrown

### Steps to reproduce

1. Create site with a dash `test-test` and another one without a dash `test`
2. Create a Global
3. Access globals cp section, choose the site with a dash
4. error is thrown

### Expected behavior

Globals should load correctly
<img width="1260" alt="Captura de pantalla 2023-01-12 a las 14 30 50" src="https://user-images.githubusercontent.com/90317987/212080466-716fac0e-e5bb-4fed-a79e-06fad5e42a95.png">

### Actual behavior

An error is thrown

<img width="1260" alt="Captura de pantalla 2023-01-12 a las 14 31 06" src="https://user-images.githubusercontent.com/90317987/212080415-cc73f045-7307-442d-8b3e-fb76a0d7c933.png">

Stack Trace
![Captura de pantalla 2023-01-12 a las 14 31 37-fullpage](https://user-images.githubusercontent.com/90317987/212080698-5f407055-ac84-490a-be4d-f983bafc3e36.png)

### Craft CMS version

3.7.63.1

### PHP version

8.1

### Operating system and version

Debian

### Database type and version

Mysql

### Installed plugins and versions

-

**Comment by eduardoalonsoalbella** [comment]
Dashes are not allowed in `sites`

**Comment by brandonkelly** [comment]
@eduardoalonsoalbella How did you create a site with a dash to begin with?

**Comment by eduardoalonsoalbella** [comment]
@brandonkelly updating the config files 😅

**Comment by brandonkelly** [comment]
Ah ok. Yeah only alphanumeric and `_` characters are allowed in handles.

**Comment by eduardoalonsoalbella** [comment]
Yes, that's why I discarded this bug report, since "our" behavior is not how craft is supposed to work 🙃
